File: android/network/SafeConnectApi.kt
package com.safeconnect.app.network

import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.GET
import retrofit2.http.Path

data class CreateUserRequest(
    val name: String,
    val email: String? = null
)

data class UserResponse(
    val id: String,
    val name: String,
    val email: String?
)

enum class CheckInType {
    OK, NEED_TO_TALK
}

data class CheckInRequest(
    val type: CheckInType
)

data class CheckInResponse(
    val id: String,
    val user_id: String,
    val timestamp: String,
    val type: CheckInType
)

interface SafeConnectApi {

    @POST("users")
    suspend fun createUser(@Body body: CreateUserRequest): UserResponse

    @POST("users/{id}/checkins")
    suspend fun createCheckIn(
        @Path("id") userId: String,
        @Body body: CheckInRequest
    ): CheckInResponse

}

File: ios/network/SafeConnectAPI.swift
import Foundation

class SafeConnectAPI {

    static let shared = SafeConnectAPI()
    private let baseURL = "http://localhost:8000/"

    func createUser(name: String) async throws -> UserResponse {
        let url = URL(string: "\(baseURL)users")!
        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")

        let body = CreateUserRequest(name: name, email: nil)
        request.httpBody = try JSONEncoder().encode(body)

        let (data, _) = try await URLSession.shared.data(for: request)
        return try JSONDecoder().decode(UserResponse.self, from: data)
    }

    func sendCheckIn(userId: String, type: String) async throws -> CheckInResponse {
        let url = URL(string: "\(baseURL)users/\(userId)/checkins")!
        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")

        let body = CheckInRequest(type: type)
        request.httpBody = try JSONEncoder().encode(body)

        let (data, _) = try await URLSession.shared.data(for: request)
        return try JSONDecoder().decode(CheckInResponse.self, from: data)
    }
}